situp-plugins/common/src/main/java/com/amazon/situp/plugins/processor/StringProcessor.java

```diff
@@ -25,19 +27,16 @@
      * @param pluginSetting instance with metadata information from pipeline pluginSetting file.
      */
     public StringProcessor(final PluginSetting pluginSetting) {
-        this();
-    }
-
-    public StringProcessor() {
-
+        this.toUpperCase = pluginSetting.getBooleanOrDefault("upper_case", true);
     }
 
     @Override
     public Collection<Record<String>> execute(final Collection<Record<String>> records) {
         final Collection<Record<String>> modifiedRecords = new ArrayList<>(records.size());
         for (Record<String> record : records) {
             final String recordData = record.getData();
-            modifiedRecords.add(new Record<>(recordData.toUpperCase()));
+            final String newData = toUpperCase? recordData.toUpperCase() : recordData.toLowerCase();
```

Comment:
Nit: toUpperCase<space>?

Reply:
Also, if upper_case is not set - is it needed to do lower_case ? [I don't have a problem but if we expand this processor with other attributes and I don't want case modification at all - it is not possible right;]

Reply:
change variable name to `upperCase`?

Reply:
say if we add other attributes (features) later - we are doing either upper case or lower case -> there is no way for me to use this processor without modifying the case; should we not do `recordData.toLowerCase();` if `toUppercase is false`. What do u think ?

Reply:
> Also, if upper_case is not set - is it needed to do lower_case ? [I don't have a problem but if we expand this processor with other attributes and I don't want case modification at all - it is not possible right;]

For the item laid out in #159 , this is the desired behavior. @kowshikn Shall we leave more customized string conversion to future work?

Reply:
I am fine with it for now - but we renamed it to String_converter when its basically case converter - but its fine if we later are going to add features to this - in that case I think we should leave an issue for future

Reply:
Discussed with Kowshik offline, for now it is only a sample processor. Will customize it later.